### PR TITLE
feat(cli/migrate): migrate existing v1 plugins

### LINKED ISSUE #9400: [bug] Tauri auto-migration util fails
### Describe the bug

[Migration util](https://beta.tauri.app/guides/upgrade-migrate/from-tauri-1/#automated-migration) is not working:

```
npm run tauri migrate

> sigma-file-manager-v2@2.0.0-alpha.1 tauri
> tauri migrate

    Info Replacing `@tauri-apps/api/tauri` with `@tauri-apps/api/core` on C:\Users\aleks\data\projects\sigma-file-manager\src\stores\storage\workspaces.ts
    Info Installing Cargo dependency "tauri-plugin-http"...
    Updating crates.io index
      Adding tauri-plugin-http v2.0.0-beta.4 to dependencies.
             Features:
             - blocking
             - brotli
             - cookies
             - default-tls
             - deflate
             - gzip
             - http3
             - json
             - multipart
             - native-tls
             - native-tls-alpn
             - native-tls-vendored
             - rustls-tls
             - rustls-tls-manual-roots
             - rustls-tls-native-roots
             - rustls-tls-webpki-roots
             - socks
             - stream
             - trust-dns
             - unsafe-headers
    Blocking waiting for file lock on package cache
    Updating crates.io index
error: invalid character `{` in package name: `tauri-plugin-{{name}}`, characters must be Unicode XID characters (numbers, `-`, `_`, or most letters)
 --> ..\..\..\..\.cargo\git\checkouts\plugins-workspace-fd0259fcb18f51ad\5817611\shared\template\Cargo.toml:2:8
  |
2 | name = "tauri-plugin-{{name}}"
  |        ^^^^^^^^^^^^^^^^^^^^^^^
  |
error: failed to select a version for `tauri`.
    ... required by package `sigma-file-manager-v2 v2.0.0-alpha.1 (C:\Users\aleks\data\projects\sigma-file-manager\src-tauri)`
versions that meet the requirements `^2.0.0-beta` are: 2.0.0-beta.14, 2.0.0-beta.13, 2.0.0-beta.12, 2.0.0-beta.11, 2.0.0-beta.10, 2.0.0-beta.9, 2.0.0-beta.8, 2.0.0-beta.7, 2.0.0-beta.6, 2.0.0-beta.5, 2.0.0-beta.4, 2.0.0-beta.3, 2.0.0-beta.2, 2.0.0-beta.1, 2.0.0-beta.0

the package `tauri` links to the native library `Tauri`, but it conflicts with a previous package which links to `Tauri` as well:
package `tauri v2.0.0-beta.14`
    ... which satisfies dependency `tauri = "^2.0.0-beta"` of package `sigma-file-manager-v2 v2.0.0-alpha.1 (C:\Users\aleks\data\projects\sigma-file-manager\src-tauri)`
Only one package in the dependency graph may specify the same links value. This helps ensure that only one copy of a native library is linked in the final binary. Try to adjust your dependencies so that only one package uses the `links = "Tauri"` value. For more information, see https://doc.rust-lang.org/cargo/reference/resolver.html#links.

the package `sigma-file-manager-v2` depends on `tauri`, with features: `window-set-content-protected` but `tauri` does not have these features.


failed to select a version for `tauri` which could resolve this conflict
    Error Failed to install Cargo dependency
```

### Reproduction

Repo: https://github.com/aleksey-hoffman/sigma-file-manager/tree/v2

Following [auto-migration guide](https://beta.tauri.app/guides/upgrade-migrate/from-tauri-1/#automated-migration): 
```
npm install @tauri-apps/cli@next
npm run tauri migrate
```

### Expected behavior

To auto migrate project from v1 to v2

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.22631 X64
    ✔ WebView2: 123.0.2420.81
    ✔ MSVC: Visual Studio Community 2022
    ✔ rustc: 1.77.1 (7cf61ebde 2024-03-27)
    ✔ cargo: 1.77.1 (e52e36006 2024-03-26)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.12.1
    - pnpm: 8.8.0
    - npm: 10.5.0

[-] Packages
    - tauri [RUST]: 1.6.0
    - tauri-build [RUST]: 1.5.1
    - wry [RUST]: 0.24.7
    - tao [RUST]: 0.16.7
    - tauri-cli [RUST]: 2.0.0-alpha.6
    - @tauri-apps/api [NPM]: 1.5.3
    - @tauri-apps/cli [NPM]: 2.0.0-beta.12

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:5180/
    - framework: Vue.js
    - bundler: Vite
```

### PR DESCRIPTION
**Pull request by amrbashir**
closes #9400

### CODE REVIEW

File: tooling/cli/src/migrate/frontend.rs [hunk 1]
```diff
@@ -56,18 +47,15 @@ pub fn migrate(app_dir: &Path, tauri_dir: &Path) -> Result<()> {
...
                 "globalShortcut" => module = String::from("global-shortcut"),
                 _ => {}
               }
-              let js_plugin = format!("@tauri-apps/plugin-{module}");
-              let cargo_crate = format!("tauri-plugin-{module}");
-              new_npm_packages.push(js_plugin.clone());
-              new_cargo_packages.push(cargo_crate);
-              js_plugin
+              new_plugins.push(module);
+              format!("@tauri-apps/plugin-{module}")
```
Comment (amrbashir) [inline tooling/cli/src/migrate/frontend.rs]:
```suggestion
              let plugin = format!("@tauri-apps/plugin-{module}");
              new_plugins.push(module);
              plugin
```

**Comment by FabianLars** [comment]
I didn't test it but fs-extra and fs-watch were merged into fs and i think the code logic would fail for that, right?

On that note, i still plan to rename upload to http-extra so it'd be nice if we could have that in mind too.

**Comment by amrbashir** [comment]
converting to draft as I would like #10198 to get merged first before I re-iterate on a better approach that modifies `Cargo.toml` and `package.json` directly without relying on `cargo add` or `npm add`

**Comment by amrbashir** [comment]
superseded by #10656

**Pull request closed without merging**